edgepi-dac.js

```diff
@@ -2,61 +2,67 @@ module.exports = function (RED) {
   const rpc = require("@edgepi-cloud/edgepi-rpc");
 
   function DacNode(config) {
-    // Create new node instance with user config
     RED.nodes.createNode(this, config);
     const node = this;
-    const ipc_transport = "ipc:///tmp/edgepi.pipe";
-    const tcp_transport = `tcp://${config.tcpAddress}:${config.tcpPort}`;
-    const transport =
-      config.transport === "Network" ? tcp_transport : ipc_transport;
-    const voltage = parseFloat(config.voltage);
-    const channel = rpc.DACChannel[config.dacChannel];
+    let voltage = config.voltage;
+    let gain = config.gain;
+    let channel = config.channel;
 
-    const dac = new rpc.DacService(transport);
+    initializeNode(config).then((dac) => {
+      node.on("input", async function (msg, send, done) {
+        node.status({ fill: "green", shape: "dot", text: "input received" });
+        try {
+          gain = msg.gain ?? gain;
+          voltage = msg.payload ?? voltage;
+          channel = msg.channel ?? channel;
 
-    //  TODO: potentially update if condition based on pos values of dac
-    if (dac) {
-      console.log(config.gainOption);
-      console.log(config.voltage);
-      console.log(config.transport);
-      console.log(config.tcpPort);
-      console.info("DAC node initialized on:", transport);
-      node.status({ fill: "green", shape: "ring", text: "dac initialized" });
-    } else {
-      node.status({
-        fill: "red",
-        shape: "ring",
-        text: "dac failed to initialize.",
+          if ((gain && voltage > 10) || voltage < 0 || (!gain && voltage > 5)) {
+            throw new Error(
+              "Voltage being written is outside the valid range."
+            );
+          }

```

Comment:
how does `gain && voltage` evaluate to a number?

Reply:
Those are evaluated separately. Order of operations in that part: `gain`, `voltage > 10`, `gain && voltage > 10`.